torch/tensortypes/nestedtensor/nested.py

```diff
@@ -0,0 +1,215 @@
+                    for tensor in nested_tensor.tensors:
+                        tensor.grad_fn(*args, **kwargs)
+                return _func
+            return create_grad_fn(self)
+        if attr == 'grad':
+            grads = []
+            for tensor in self.tensors:
+                grads.append(tensor.grad)
+            if all(grad is None for grad in grads):
+                return None
+            else:
+                return NestedTensor(grads)
+        if attr == 'data':
+            data = []
+            for tensor in self.tensors:
+                data.append(tensor.data)
+            return NestedTensor(data)
+        return super().__getattribute__(attr)
+
+    def __len__(self):
+        return len(self.tensors)
+
+    def __bool__(self):
+        raise NotImplementedError(
+            "This has not been covered by NestedTensor 0.0.1")
+
+    def __str__(self):
+        tensors = self.unbind()
+        result = "nestedtensor([\n"
+        for tensor in tensors:
+            result += "  " + tensor.__str__() + "\n"
+        result += "])"
+        return result
+
+    def __repr__(self):
+        tensors = self.unbind()
+        result = "nestedtensor([\n"
+        for tensor in tensors:
+            result += "  " + tensor.__repr__() + "\n"
+        result += "])"
+        return result
+
+    def is_empty(self):
+        # This condition can never be true, since we disallow an empty list for now.
+        raise ValueError("self.tensors cannot be empty under current constraints.")
+
+    def __loop__apply(self, fn):
+        tensors = []
+        for tensor in self.tensors:
+            tensors.append(fn(tensor))
+        return tensors
+
+    def nested_size(self):
+        tensors = self.unbind()
+        sizes = []
+        for tensor in tensors:
+            sizes.append(tensor.size())
+        return tuple(sizes)
+
+    # TODO: Not covered by RFC! NestedTensor 0.0.2 will talk about reductions.
+    def all(self):
+        ret = True
+        for tensor in self.tensors:
+            ret = ret and tensor.all()
+        return ret
+
+    # TODO: Not covered by RFC! NestedTensor 0.0.2 will talk about reductions.
+    def any(self):
+        ret = False
+        for tensor in self.tensors:
+            ret = ret or tensor.any()
+        return ret
+
+    # TODO: Not covered by RFC! NestedTensor 0.0.2 will talk about reductions.
+    def sum(self):
+        sums = []
+        # We currently assume len(self.tensors) is always non-zero
+        for tensor in self.tensors:
+            sums.append(tensor.sum())
+        return torch.stack(sums).sum()
+
+    # Tensor ops
+    def detach(self):
+        return NestedTensor(self.__loop__apply(lambda x: x.detach()))
+
+    def detach_(self):
+        return NestedTensor(self.__loop__apply(lambda x: x.detach_()))
+
+    def backward(self, *args, **kwargs):
+        self.tensors = self.__loop__apply(lambda x: x.backward(*args, **kwargs))
```

Comment:
This looks very suspicious.  `Tensor.backward` doesn't even return anything!

Reply:
It's not described in the operator contract that it is to return something https://pytorch.org/docs/master/tensors.html#torch.Tensor.backward

Reply:
So why are you assigning `self.tensors` with the result of calling backward? That sounds like a one way ticket to broken internal invariants